### ISSUE TITLE
tsql commands require GO statement - otherwise nothing is returned

### ISSUE THREAD

**Issue by powelleric** [FreeTDS/freetds#187]
In a previous issue, @freddy77 [mentioned](https://github.com/FreeTDS/freetds/issues/161#issuecomment-345859035) that tsql was changed to no longer require a GO statement at the end of each query.

I have pulled the most recent code from the master branch but this feature does not seem to be working; I still need GO otherwise the result is empty.

Thanks a bunch for your help!

**Comment by freddy77** [comment]
Are you using it interactively?
Are you ending your line with a end-of-line?
Should work with both interactive and not. Tested with "echo" and worked.

**Comment by powelleric** [comment]
Thank you for your reply.  Yes, my sql is definitely ended with a newline character.  Below I have pasted exactly what I am doing.  You can see that the first time I call tsql I get nothing.  After appending GO, I get the results.

Am I doing something wrong?
```

$ cat tmp.sql 
select top 10 CustomerId from Customer

$ ~/lib/freetds/src/apps/tsql -U user@server -S server -P 'pass' -D dev -o qh < tmp.sql
$ echo "GO" >> tmp.sql
$ cat tmp.sql
select top 10 ExternalSampleAliquot from Sample

GO
$ ~/lib/freetds/src/apps/tsql -U user@server -S server -P 'pass' -D dev -o qh < tmp.sql
201502830
21437
201304842
201505744-01
$ ~/lib/freetds/src/apps/tsql -C
Compile-time settings (established with the "configure" script)
                            Version: freetds v1.1.dev.20171009
             freetds.conf directory: /usr/local/etc
     MS db-lib source compatibility: no
        Sybase binary compatibility: no
                      Thread safety: yes
                      iconv library: no
                        TDS version: auto
                              iODBC: no
                           unixodbc: yes
              SSPI "trusted" logins: no
                           Kerberos: no
                            OpenSSL: yes
                             GnuTLS: no
                               MARS: yes
```

**Comment by freddy77** [comment]
Can you try this patch and see what it print?

```
diff --git a/src/apps/tsql.c b/src/apps/tsql.c
index 1359f456..8d4506c2 100644
--- a/src/apps/tsql.c
+++ b/src/apps/tsql.c
@@ -877,8 +877,11 @@ main(int argc, char **argv)
                sprintf(prompt, "%d> ", ++line);
                s = tsql_readline(QUIET ? NULL : prompt);
                if (s == NULL) {
-                       if (buflen)
+                       printf("end of file\n");
+                       if (buflen) {
+                               printf("executing query\n");
                                do_query(tds, mybuf, global_opt_flags);
+                       }
                        break;
                }


```

**Comment by powelleric** [comment]
Sorry for the delayed response.  I had an unfortunate issue that caused my main development server to become unreachable.  I am still recovering from that and so I have been working on my OSX laptop instead of my old Ubuntu 16.04 server.

The good news is that tsql is definitely working without any 'Go' statement.  I did **NOT** use the patch you provided above;  I just pulled a fresh copy from Github today and everything was fine. 

(If you did not commit the patch, then I cannot explain why it is working now but not on the Ubuntu server.)

Anyway, thank you for your help.  I really, really appreciate it!
